zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreter.java

```diff
@@ -76,17 +83,46 @@ public RemoteInterpreter(Properties property,
     this.connectTimeout = connectTimeout;
     this.maxPoolSize = maxPoolSize;
     this.remoteInterpreterProcessListener = remoteInterpreterProcessListener;
+    this.applicationEventListener = appListener;
+  }
+
+
+  /**
+   * Connect to existing process
+   */
+  public RemoteInterpreter(Properties property,
+                           String noteId,
+                           String className,
+                           String host,
+                           int port,
+                           int connectTimeout,
+                           int maxPoolSize,
+                           RemoteInterpreterProcessListener remoteInterpreterProcessListener,
+                           ApplicationEventListener appListener) {
+    super(property);
+    this.noteId = noteId;
+    this.className = className;
+    initialized = false;
+    this.host = host;
+    this.port = port;
+    this.connectTimeout = connectTimeout;
+    this.maxPoolSize = maxPoolSize;
+    this.remoteInterpreterProcessListener = remoteInterpreterProcessListener;
+    this.applicationEventListener = appListener;
   }
 
+
+  // VisibleForTesting
   public RemoteInterpreter(Properties property,
-      String noteId,
-      String className,
-      String interpreterRunner,
-      String interpreterPath,
-      String localRepoPath,
-      Map<String, String> env,
-      int connectTimeout,
-      RemoteInterpreterProcessListener remoteInterpreterProcessListener) {
+                           String noteId,
```

Comment:
I don't have preference of code formatting, but my IDE seems it does. 
I'll restore the style.